Hotfix/documentation configuration (#19)
qualabs/video-headline: CONFIGURATION.md

# Configuration

#### Creating a superuser

> *Is assumed that by this time all images are running on docker. (check: - [Running the application in local environment](#running-the-application-in-local-environment))

Reviewer: Which images?
I would write something about after executing docker commands.

Author: done.

> 1. Create a superuser for admin access running `python manage.py createsuperuser`.

Reviewer 2: createsuperuser for the local enviroment is also in the CONTRIBUTING.md, should be on both files?

Reviewer 3: In my opinion I would change this for: `In case you want to create another superuser run: python manage.py createsuperuser`

Author: the create super user documentation is part of the acceptance criteria, in that sense idk if the create super user documentation is going to remain on other files.

Reviewer 4: I would state "Creating another superuser" to be clear there is one already created and this is advanced configuration

> 2. Go to `http://localhost:8010/admin` and log in with the superuser credentials.

Reviewer: ```suggestion
2. Go to `http://localhost:8010/admin` and sign in with the superuser credentials.
```

Author: added.

> 2. Go to `http://localhost:8010/admin` and log in with the superuser credentials.

Reviewer 5: should we also include instructions for creating a super user on deployed instances?

#### AWS Services Configuration in the Admin Web

> *In order to apply this configurations, login to the admin web is needed.

Reviewer: ```suggestion
*In order to apply the following configurations, signing in to admin web is needed.
```

Author: done

> 1. **CloudFront Configuration:** Inside the Global configuration, apply the settings located in configuration/configuration.samples/cloud_front_configuration.sample on the CloudFront Configuration input.

Reviewer 5: ```suggestion
1. **CloudFront Configuration:** Inside `Global Configuration`, apply the settings located in `configuration/configuration.samples/cloud_front_configuration.sample` on the `CloudFront Configuration` input.
```

Author: done.

> 2. **MediaConvert Configuration:** Create a `MediaConvert Configuration` named `default`, using the settings found in `configuration/configuration.samples/media_convert_configuration.sample`.

Reviewer: Why named default?

Author: to be honest this was inherited from the original documentation, actually yes the name can be chosen, so i changed this

> 3. **MediaLive Configuration:** Create `MediaLive Configuration`, also named `default`, and apply the following configurations:

Reviewer: Why named default?

Author: changed

> 3. **MediaLive Configuration:** Create `MediaLive Configuration`, also named `default`, and apply the following configurations:

Reviewer 5: ```suggestion
3. **MediaLive Configuration:** Create a `MediaLive Configuration`, also named `default`, and apply the following configurations:
```

Author: added.

   - **Input Configuration:** `configuration/configuration.samples/media_live_input_attachments.sample`.
   - **Destination Configuration:** `configuration/configuration.samples/media_live_destinations.sample`.
   - **Encoder Configuration:** `configuration/configuration.samples/media_live_encoder_settings_economic.sample`.

   ![¡](docs/aws-services-configuration.png)

#### Creating a Plan

> *On the "Organizations and channels" menu choose "Plans" and click on the "add plan" button.

Reviewer 5: ```suggestion
*On the "Organizations and Channels" menu choose "Plans" and click on "Add Plan".
```

Author: done.

![¡](docs/orgs-and-channels.png)

1. **Business name:** Add a name for the plan.

> 2. **Video transcoding configuration:** From the dropdown choose the configuration needed.

Reviewer: Dropdown is more a programmer expression

Reviewer 5: also we did name things on previous step, it would be good to use those names, for example `From the selector choose the "default" configuration (or any other name you may have used)`. Apply this suggestion to other steps

Author: added.

> 2. **Audio transcoding configuration:** From the dropdown choose the configuration needed.

Reviewer: Dropdown is more a programmer expression

Reviewer 3: instead of dropdown we can use: selection menu OR option list

Author: added the previous suggestion

> 3. **MediaLive Configuration:** From the dropdown choose the configuration needed.

Reviewer: Dropdown is more a programmer expression

Reviewer 3: instead of dropdown we can use: selection menu OR option list

Author: added the previous suggestion

![¡](docs/plan.png)

#### Adding a AWS account

*On the "Organizations and channels" menu choose "AWS account" and click on the "add AWS account" button.

1. **Name:** Add a name for the account.
2. **Access Key:** The Access Key for the API user (The user was created in the AWS Configuration section, and the credentials are available in AWS Secrets Manager).
3. **Secret Access Key:** The Secret Access Key for the API user (The user was created in the AWS Configuration section, and the credentials are available in AWS Secrets Manager).
4. **Region:** The region in which your AWS Account is deployed.
5. **MediaConvert Role:** On the AWS IAM service, on the "roles" section there is a search bar, which can be used to get the MediaConvertRole value.
6. **MediaConvert Endpoint URL** This value is on the "account" section of the Media Convert service in AWS.
7. **MediaLive Role:** On the AWS IAM service, on the "roles" section there is a search bar, which can be used to get the MediaLiveAccessRole value.
8. **Account Id:** Your AWS account number.

![¡](docs/aws-account.png)

#### Creating an Organization

> *On the "Organizations and channels" menu choose "Organization" and click on the "add organization" button.

Reviewer 5: ```suggestion
*On the "Organizations and Channels" menu choose "Organization" and click on the "Add Organization" button.
```

1. **Name:** Add a name for your organization. This step will create an s3 bucket with the name of the organization, the name must follow the bucket naming rules: https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html .
2. **Plan:** From the dropdown choose the plan previously created.

> 3. **AWS Account:** From the dropdown choose the AWS Account previously created.

Reviewer 3: we should also change the dropdown word here.

![¡](docs/organization.png)

#### Associating an Organization to a user

*On the "Authentication and Authorization" menu choose "Users" and click on the username to edit.

1. **Organization:** Assign the newly created Organization to the superuser to enable authentication via the User web.

![¡](docs/auth.png)

#### Periodic Tasks Configuration

> *All kind of tasks can be set to run periodically. This can be done on the "Periodic Tasks" menu.

Reviewer 5: Can we provide examples on what "tasks" we are talking about?

Author: added picture with options

1. **Clocked:** Here an specific date and hour can be set to run a task.
2. **Crontabs** Cron configuration, the cron can be configured to run at specific months of the year, specific days of the week, and even specify down to the minute when a particular task should be executed.
3. **Intervals:** Set intervals to wait before running the same task again, it can be set in days, hours, minutes, seconds, and microseconds.

> 4.**Periodic Tasks** Here a Name must be set for the task, and choose a tasl from "Task(registered)" dropdown. Inside the Schedule menu we can choose from the previously created options when we want the task to be executed.

Reviewer: ```suggestion
4.**Periodic Tasks** Here a Name must be set for the task, and choose a task from "Task(registered)" dropdown. Inside the Schedule menu we can choose from the previously created options when we want the task to be executed.
```

Reviewer: I would say something like : 
On schedule menu, we can choose previous created options so as to schedule the task.

Reviewer 3: we should also change the dropdown word here.

Author: done.

> 5. **Solar Events** The task can be executed on a specific zone, this can be set by adding "Latitue" and "Longitude" and selecting one of the solar events from the dropdown menu.

Reviewer: ```suggestion
5. **Solar Events:** The task can be executed in a specific zone, this can be set by adding "Latitude" and "Longitude" and selecting one of the solar events from the dropdown menu.
```

Reviewer 3: we should also change the dropdown word here.

Author: added

![¡](docs/periodic-tasks.png)

### Custom CSS Configuration for the Player in an Organization

> 1. Have a URL ready with a CSS file to test (not a local path).

Reviewer: What are these? Prerrequisites?

2. In the Organization, add the following line in the organization configuration field:

```
  "playerCustomCss":"https://your-css-url/cssFile.css"
```

> If you do not have a CSS URL, you can serve a folder using the npx serve command: 1. Navigate to the folder containing the CSS file you want to use. 2. Run the command 'npx serve .'. 3. Open the server from the browser (probably http://localhost:5000), find the file, and copy the URL.

Reviewer: What do you mean by serve a folder?

Reviewer: Also , add some spaces for instructions

# Configuration

#### Creating another superuser

*This should be executed after executing all docker commands. (check: - [Running the application in local environment](#running-the-application-in-local-environment))

> 1. Create a superuser for admin access running `python manage.py createsuperuser`.

Reviewer 6: shouldn't it be like this?
1. Run `docker exec -it video-hub bash` to access the video-hub container.
2. Create a superuser for admin access running `python manage.py createsuperuser`

3. **MediaLive Configuration:** From the menu choose the configuration needed.

![¡](docs/plan.png)

#### Adding a AWS account

*On the `Organizations and Channels` menu choose `AWS account` and click on the `add AWS account` button.

1. **Name:** Add a name for the account.
2. **Access Key:** The Access Key for the API user (The user was created in the AWS Configuration section, and the credentials are available in AWS Secrets Manager).
3. **Secret Access Key:** The Secret Access Key for the API user (The user was created in the AWS Configuration section, and the credentials are available in AWS Secrets Manager).
4. **Region:** The region in which your AWS Account is deployed.
5. **MediaConvert Role:** On the AWS IAM service, on the "roles" section there is a search bar, which can be used to get the MediaConvertRole value.
6. **MediaConvert Endpoint URL** This value is on the "account" section of the Media Convert service in AWS.
7. **MediaLive Role:** On the AWS IAM service, on the "roles" section there is a search bar, which can be used to get the MediaLiveAccessRole value.

> 8. **Account Id:** Your AWS account number.

Reviewer 6: should explain where find the account id ?

1. Create a superuser for admin access running `python manage.py createsuperuser`.
2. Go to `http://localhost:8010/admin` and sign in with the superuser credentials.

#### AWS Services Configuration in the Admin Web

*In order to apply the following configurations, signing in to admin web is needed.

1. **CloudFront Configuration:** Inside `Global configuration`, apply the settings located in `configuration/configuration.samples/cloud_front_configuration`. on the `CloudFront Configuration` input.
2. **MediaConvert Configuration:** Create a `MediaConvert Configuration` and add a name to it, using the settings found in `configuration/configuration.samples/media_convert_configuration.sample`.
3. **MediaLive Configuration:** Create a `MediaLive Configuration`, add a name to it, and apply the following configurations:
   - **Input Configuration:** `configuration/configuration.samples/media_live_input_attachments.sample`.
   - **Destination Configuration:** `configuration/configuration.samples/media_live_destinations.sample`.

>    - **Encoder Configuration:** `configuration/configuration.samples/media_live_encoder_settings_economic.sample`.

Reviewer 4: Here I'd explain that we have those samples (and we already created the default ones with those sample files, but if you want to modify those files you can then find them under configuration/x... file and expand/change those "default" ones to then create more configurations
